addons/website_sale/tests/test_main_controller.py

```diff
@@ -0,0 +1,97 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.tests import tagged
+from odoo.http import request
+
+from odoo.addons.base.tests.common import HttpCaseWithUserPortal
+from odoo.addons.payment.tests.common import PaymentCommon
+from odoo.addons.payment.models.payment_provider import PaymentProvider
+
+from unittest.mock import patch, MagicMock
+
+
+@tagged('post_install', '-at_install')
+class TestPaymentProviderVisibility(HttpCaseWithUserPortal, PaymentCommon):
+
+    def setUp(self):
```

Comment:
if the setup is only meant for one test, it should be in the test directly, there is no need to define a `setUp` method (and it's a bad practice to use `setUp`, `setUpClass` should be used instead so that the setup is only done once by test class).